pyomo/core/expr/numvalue.py

```diff
@@ -218,16 +267,35 @@ def potentially_variable(obj):
     if obj.__class__ in native_types:
         return False
     try:
-        return obj._potentially_variable()
+        return obj.is_potentially_variable()
+    except AttributeError:
+        pass
+    return as_numeric(obj).is_potentially_variable()
+
+def polynomial_degree(obj):
+    """
+    A utility function that returns an integer 
+    that indicates the polynomial degree for an
+    object. boolean indicating
+    """
+    if obj.__class__ in native_types:
+        return 0
+    try:
+        return obj.polynomial_degree()
     except AttributeError:
         pass
-    return as_numeric(obj)._potentially_variable()
+    return as_numeric(obj).polynomial_degree()
 
 # It is very common to have only a few constants in a model, but those
 # constants get repeated many times.  KnownConstants lets us re-use /
 # share constants we have seen before.
 KnownConstants = {}
 
+def update_KnownConstants(obj, val):
+    if len(KnownConstants) < 100:
+        KnownConstants[obj] = val
```

Comment:
What's the point of the "100" magic number?

Reply:
Agreed, but just caching the first 100 is also probably bad. Plus that is an unlabeled embedded constant. 

Separately, @whart222 and I discussed whether we needed the KnownConstants at all...